lib/monads/many.rb

```diff
@@ -11,7 +16,7 @@ def and_then(&block)
     end
 
     def self.from_value(value)
-      Many.new([value])
+      Many.new(value.respond_to?(:to_ary) ? value.to_ary || [value] : [value])
```

Comment:
I think this is (more or less) what `Kernel#Array` does. Any reason we can’t have `Many.new(Array(value))` here instead?